pandas/tests/io/test_common.py

```diff
@@ -119,10 +121,11 @@ def test_infer_compression_from_path(self, extension, expected, path_type):
     @pytest.mark.parametrize("path_type", [str, CustomFSPath, Path])
     def test_get_handle_with_path(self, path_type):
         # ignore LocalPath: it creates strange paths: /absolute/~/sometest
-        filename = path_type("~/sometest")
-        with icom.get_handle(filename, "w") as handles:
-            assert os.path.isabs(handles.handle.name)
-            assert os.path.expanduser(filename) == handles.handle.name
+        with tempfile.NamedTemporaryFile(dir=os.path.expanduser("~/")) as tmp:
```

Comment:
use tm.ensure_clean

Reply:
I agree, but current implementation of `tm.ensure_clean` doesn't support custom directories. Should we then first extend that function?

Reply:
ok i guess this is fine. can you use pathlib.Path instead